src/components/ShippingLabels/ShippingLabel.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import supabase, { Schema } from "@/supabase";
+import ShippingLabelsButton from "@/components/ShippingLabels/ShippingLabelsButton";
+import { ParcelClients } from "@/components/ShippingLabels/ShippingLabelsPdf";
+
+const formatDatetime = (datetimeString: string | null, isDatetime: boolean): string => {
+    if (datetimeString === null) {
+        return "No recorded date";
+    }
+
+    const dateOptions: Intl.DateTimeFormatOptions = {
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "2-digit",
+        minute: "2-digit",
+    };
+
+    const formattedDate = isDatetime
+        ? new Date(datetimeString).toLocaleString([], dateOptions)
+        : new Date(datetimeString).toLocaleDateString();
+
+    return formattedDate;
+};
+
+const format1DTo2DArray = (
+    inputArray: ParcelClients[],
+    elementsPerSubarray: number
+): ParcelClients[][] => {
+    const result = [];
+
+    for (let index = 0; index < inputArray.length; index += elementsPerSubarray) {
+        result.push(inputArray.slice(index, index + elementsPerSubarray));
+    }
+
+    return result;
+};
+
+// TODO: make the function query only required data from selected rows in client page
```

Comment:
Is this a TODO for you? Or a later ticket?

Reply:
For a later ticket. For whoever is the one who puts this component in the client page.